app/controllers/tag_controller.rb

```diff
@@ -567,4 +538,25 @@ def order_string
       params[:order] == "asc" ? "name ASC" : "name DESC"
     end
   end
+
+  def fetch_counts
+    # Enhancement #6306 - Add counts to `by type` dropdown on tag pages
+    @counts = {}
+    @counts[:posts] = Node.for_tagname_and_type(params[:id], 'note', wildcard: @wildcard).where('node.nid NOT IN (?)', @qids).count
+    @counts[:questions] = Node.for_tagname_and_type(params[:id], 'note', question: true, wildcard: @wildcard).where('node.nid IN (?)', @qids).count
+    @counts[:wiki] = Node.for_tagname_and_type(params[:id], 'page', wildcard: @wildcard).count
```

Comment:
Now the counts are using the same queries and logic as the rest of the controller.